File: AtlanticClojure/Metrics.clj
( defstruct Metric :name )
( defstruct MetricValue :tag )
( defstruct StringMetricValue :value :tag )
( defstruct BooleanMetricValue :value :tag )
( defstruct IntegerMetricValue :value :tag )
( defstruct DoubleMetricValue :value :tag )

File: AtlanticOsloM/Metrics.m
module Metrics {

	type Metric {
		Name_Att : Text;
		Values_ContRef : MetricValue*;
	};

	type MetricValue {
		Tag_Att : Text;
	};

	type StringMetricValue : MetricValue {
		Value_Att : Text;
	};

	type BooleanMetricValue : MetricValue {
		Value_Att : Logical;
	};

	type IntegerMetricValue : MetricValue {
		Value_Att : Integer32;
	};

	type DoubleMetricValue : MetricValue {
		Value_Att : Double;
	};
}